src/Factory/CreditMemoFactory.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Sylius\RefundPlugin\Factory;
+
+use Sylius\RefundPlugin\Entity\CreditMemoChannel;
+use Sylius\RefundPlugin\Entity\CreditMemoInterface;
+use Sylius\RefundPlugin\Entity\CustomerBillingDataInterface;
+use Sylius\RefundPlugin\Entity\ShopBillingData;
+
+class CreditMemoFactory implements CreditMemoFactoryInterface
+{
+    /** @var string */
+    protected $className;
+
+    public function __construct(string $className)
+    {
+        $this->className = $className;
+    }
+
+    public function createForData(
+        string $id,
+        string $number,
+        string $orderNumber,
+        int $total,
+        string $currencyCode,
+        string $localeCode,
+        CreditMemoChannel $channel,
+        array $units,
+        string $comment,
+        \DateTimeInterface $issuedAt,
+        CustomerBillingDataInterface $from,
+        ?ShopBillingData $to
+    ): CreditMemoInterface {
+        return new $this->className(
+            $id,
+            $number,
+            $orderNumber,
+            $total,
+            $currencyCode,
+            $localeCode,
+            $channel,
+            $units,
+            $comment,
+            $issuedAt,
+            $from,
+            $to
+        );
+    }
+
+    public function createNew()
+    {
+        throw new \RuntimeException();
```

Comment:
the factory has to implement the `FactoryInterface::createNew` although its not possible to create a `CreditMemo` without constructor parameter.

If the factory does not the ResouceController throws an exception.